generator/tests/test_other_library/run_other_library_tests.py

```diff
@@ -20,231 +20,289 @@
 
 
 ##############################################################################
-# Specific generation functions
+# Utility functions
 
+def common_set_up(filename):
+    """
+    Perform common setting-up steps.
 
-def generate_new_cpp_header(filename, num):
+    :param filename: name of XML file to parse.
+    :return: big dictionary structure obtained from the parsing.
+    """
     parser = ParseXML.ParseXML(filename)
     ob = parser.parse_deviser_xml()
     for wc in ob['baseElements']:
         strFunctions.prefix_classes(wc)
     for working_class in ob['baseElements']:
-        if working_class['name'] == global_variables.document_class:
+        if working_class['name'] == gv.document_class:
             working_class['document'] = True
+    return ob
+
+
+def go_into_new_directory():
+    """
+    Create the new ./temp directory if required. Go into it.
+    """
+    os.chdir('./temp')
+    new_dir = gv.language
+    if not os.path.isdir(new_dir):
+        os.mkdir(new_dir)
+    os.chdir(new_dir)
+
+
+##############################################################################
+# Specific generation functions
+
+
+def generate_new_cpp_header(filename, num):
+    """
+    Generate new C++ header and code files.
+
+    :param filename: the XML file to parse.
+    :param num: TODO not too sure about this - it's a list index.
```

Comment:
index allowing code to generate for just one class rather than all

Reply:
docstring updated